File: Java(eclipse)/workspace/書籍検索アプリ/src/main/java/todo/model/BookDataManager.java
package todo.model;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

import todo.model.beans.BookDataBean;

public class BookDataManager {
	
	private CsvData csvData;
	private DatabaseData databaseData;
	
	@FunctionalInterface
	public interface FunctionPointer {
		
		// Method signatures of pointed method
		String apply(BookDataBean bookData);
	}
	
	public final FunctionPointer[] SEARCH_TARGET_ARRAY = {
		//検索は上のものから順に探す
		//本のタイトルで検索
		this::getBookName,
		//著者で検索
		this::getAuthor,
		//架空ISBNコードで検索
		this::getFictitiousISBN,
	};
	
	//----------
	//コンストラクタ
	public BookDataManager() {
		
		csvData = new CsvData();
		databaseData = new DatabaseData();
	}
	
	public BookDataBean[] loadBookDatas(String filePath) {
		
        return csvData.loadBookDatas(filePath);
    }
	
	public boolean stringRegexCheck(String target,String keyWord) {
		
		return target.matches(".*" + keyWord + ".*");
	}
	
	public String getBookName(BookDataBean bookData) {
		
		//本のタイトルで検索
		return bookData.getBookName();
	}
	
	public String getAuthor(BookDataBean bookData) {
			
		//著者で検索
		return bookData.getAuthor();
	}
	
	public String getFictitiousISBN(BookDataBean bookData) {
		
		//架空ISBNコードで検索
		return bookData.getFictitiousISBN();
	}
	
	public BookDataBean[] mergedBookDataBeanGet(String[] chkBoxDataArray) {
		
	    List<ILoadData> list = new ArrayList<>();

	    //ファイルから検索
	    if (Arrays.asList(chkBoxDataArray).contains("fromFileSearch")) {
	    	
	        list.add(csvData);
	    }
	    //データベースから検索
	    if (Arrays.asList(chkBoxDataArray).contains("fromDataBaseSearch")) {
	    	
	        list.add(databaseData);
	    }

	    // 両方 or どちらか一方でも Composite にまとめる
	    ILoadData loadData = new CompositeLoadData(list.toArray(ILoadData[]::new));
	    return loadData.BookDataBeanGet();
	}
}

File: C#(VisualStudio)/WorkSpace_ASP.NETCoreMVC/書籍検索アプリ_移植元Java/Models/BookDataManager.cs
using System.Text.RegularExpressions;
using 書籍検索アプリ_移植元Java.Models.beans;

namespace 書籍検索アプリ_移植元Java.Models
{
    public class BookDataManager {
	
		public delegate string FunctionPointer(BookDataBean bookData);
		public readonly FunctionPointer[] SEARCH_TARGET_ARRAY;
		private readonly CsvData csvData;
		private readonly DatabaseData databaseData;
	
		//----------
		//コンストラクタ
		public BookDataManager() {
		
			SEARCH_TARGET_ARRAY = [

				//検索は上のものから順に探す
				//本のタイトルで検索
				//GetBookName,
				//著者で検索
				//GetAuthor,
				//架空ISBNコードで検索
				//GetFictitiousISBN,
			];
			csvData = new();
			databaseData = new();
		}

		public BookDataBean[] LoadBookDatas(string filePath) {
		
			return csvData.LoadBookDatas(filePath);
		}
	
		public bool StringRegexCheck(string target, string keyword)
		{
			var regex = ".*" + Regex.Escape(keyword) + ".*";
			return Regex.IsMatch(target, regex);
		}


		//public string GetBookName(BookDataBean bookData) {
		
		//	//本のタイトルで検索
		//	return bookData.BookName;
		//}
	
		//public string GetAuthor(BookDataBean bookData) {
			
		//	//著者で検索
		//	return bookData.getAuthor();
		//}
	
		//public string GetFictitiousISBN(BookDataBean bookData) {
		
		//	//架空ISBNコードで検索
		//	return bookData.getFictitiousISBN();
		//}
	
		public BookDataBean[] MergedBookDataBeanGet(string[] chkBoxDataArray)
		{
			var list = new List<ILoadData>();

			// ファイルから検索
			if (chkBoxDataArray.Contains("fromFileSearch"))
			{
				list.Add(csvData);
			}

			// データベースから検索
			if (chkBoxDataArray.Contains("fromDataBaseSearch"))
			{
				list.Add(databaseData);
			}

			// 両方 or 片方でも CompositeLoadData にまとめる
			ILoadData loadData = new CompositeLoadData(list);

			return loadData.BookDataBeanGet();
		}
	}
}